client/ayon_core/pipeline/actions.py

```diff
@@ -10,6 +13,224 @@
 from .load.utils import get_representation_path_from_context
 
 
+class LauncherActionSelection:
+    """Object helper to pass selection to actions.
+
+    Object support backwards compatibility for 'session' from OpenPype where
+    environment variable keys were used to define selection.
+
+    Args:
+        project_name (str): Selected project name.
+        folder_id (str): Selected folder id.
+        task_id (str): Selected task id.
+        folder_path (Optional[str]): Selected folder path.
+        task_name (Optional[str]): Selected task name.
+        project_entity (Optional[dict[str, Any]]): Project entity.
+        folder_entity (Optional[dict[str, Any]]): Folder entity.
+        task_entity (Optional[dict[str, Any]]): Task entity.
+
+    """
+    def __init__(
+        self,
+        project_name,
+        folder_id,
+        task_id,
+        folder_path=None,
+        task_name=None,
+        project_entity=None,
+        folder_entity=None,
+        task_entity=None
+    ):
+        self._project_name = project_name
+        self._folder_id = folder_id
+        self._task_id = task_id
+
+        self._folder_path = folder_path
+        self._task_name = task_name
+
+        self._project_entity = project_entity
+        self._folder_entity = folder_entity
+        self._task_entity = task_entity
+
+    def __getitem__(self, key):
+        if key in {"project_name", "AYON_PROJECT_NAME", "AVALON_PROJECT"}:
+            return self.project_name
+        if key == {"folder_path", "AYON_FOLDER_PATH", "AVALON_ASSET"}:
+            return self.folder_path
+        if key == {"task_name", "AYON_TASK_NAME", "AVALON_TASK"}:
+            return self.task_name
+        if key == "folder_id":
+            return self.folder_id
+        if key == "task_id":
+            return self.task_id
+        if key == "project_entity":
+            return self.project_entity
+        if key == "folder_entity":
+            return self.folder_entity
+        if key == "task_entity":
+            return self.task_entity
+        raise KeyError(f"Key: {key} not found")
+
+    def __contains__(self, key):
+        # Fake missing keys check for backwards compatibility
+        if key in {
+            "AYON_PROJECT_NAME",
+            "AVALON_PROJECT",
+            "project_entity",
+        }:
+            return self._project_name is not None
+        if key in {
+            "AYON_FOLDER_PATH",
+            "folder_id",
+            "folder_path",
+            "folder_entity",
+            "AVALON_ASSET",
+        }:
+            return self._folder_id is not None
+        if key in {
+            "AYON_TASK_NAME",
+            "task_id",
+            "task_name",
+            "task_entity",
+            "AVALON_TASK",
+        }:
+            return self._task_id is not None
+        return False
+
+    def get(self, key, default=None):
+        try:
+            return self[key]
+        except KeyError:
+            return default
```

Comment:
I feel like this is here mostly just for backwards compatibility - and I wonder, if moving forward we want to keep this. How about only implementing backwards compatibility here with deprecation warning so that we push for using the explicit property/getter methods?